resources/src/main/java/org/robolectric/res/android/ResourceString.java

```diff
@@ -76,7 +82,22 @@ public static String decodeString(ByteBuffer buffer, int offset, Type type) {
     } else {
       length = characterCount * 2;
     }
-    return new String(buffer.array(), offset, length, type.charset());
+    final ByteBuffer stringBuffer = ByteBuffer.wrap(buffer.array(), offset, length);
+    // Use normal UTF-8 and UTF-16 decoder to decode string
+    try {
+      return type.decoder().decode(stringBuffer).toString();
+    } catch (CharacterCodingException e) {
+      if (type == Type.UTF16) {
+        return null;
+      }
+    }
+    // Use CESU8 decoder to try decode failed UTF-8 string, especially modified UTF-8.
+    // See https://en.wikipedia.org/wiki/CESU-8.
```

Comment:
nit: can you link here instead:

https://source.android.com/devices/tech/dalvik/dex-format?hl=hr-HR&skip_cache=true#mutf-8

Reply:
Thanks, it's a good doc section I always skip. And it has been done.